docs/preview/logdecorations.md

```diff
@@ -0,0 +1,260 @@
+> ##[command]git checkout --progress --force e48a3009f2a0163d102423eef6ffaf7f4c2a2176
+  ##[command]git config http.https://repohere "AUTHORIZATION: bearer ***"
+```
+
+Nested grouping (no visual intendation) -
+```
+Syncing repository: SomeRepo (Git)
+Prepending Path environment variable with directory containing 'git.exe'.
+##[startgroup:git]
+##[startgroup]
+##[command]git version
+git version 2.18.0.windows.1
+##[startgroup]
+##[command]git config --get remote.origin.url
+##[startgroup]
+##[command]git clean -ffdx
+##[startgroup]
+##[command]git reset --hard HEAD
+##[startgroup]
+HEAD is now at cb1adf878a7b update swe
+##[startgroup]
+##[command]git config gc.auto 0
+##[startgroup]
+##[command]git config --get-all http.https://repohere
+##[startgroup]
+##[command]git config --get-all http.proxy
+##[startgroup]
+##[command]git -c http.extraheader="AUTHORIZATION: bearer ***" fetch --tags --prune --progress --no-recurse-submodules origin
+From https://repohere
+- [deleted] (none) -> origin/teams/some
+remote: Azure Repos
+remote:
+remote: Found 1444 objects to send. (1323 ms)
+Receiving objects: 0% (1/1444)
+...
+Resolving deltas: 100% (708/708), completed with 594 local objects.
+7d80bdb9d646..5214d0492d27 features/DraggableDashboardGrid -> origin/features/DraggableDashboardGrid
+...
+...
+##[startgroup]
+##[command]git checkout --progress --force e48a3009f2a0163d102423eef6ffaf7f4c2a2176
+Warning: you are leaving 1 commit behind, not connected to
+any of your branches:
+cb1adf878a7b Update CloudStore packages to 0.1.0-20190213.7 and Domino packages to 0.1.0-20190213.7
+If you want to keep it by creating a new branch, this may be a good time
+to do so with:
+git branch <new-branch-name> cb1adf878a7b
+HEAD is now at e48a3009f2a0 update swe
+##[startgroup]
+##[command]git config http.https://repohere "AUTHORIZATION: bearer ***"
+##[endgroup:git]
+```
+
+Multiple grouping parsed -
+```
+Syncing repository: SomeRepo (Git)
+Prepending Path environment variable with directory containing 'git.exe'.
+> Git commands
+```
+
+```
+Syncing repository: SomeRepo (Git)
+Prepending Path environment variable with directory containing 'git.exe'.
+v Git commands
+> ##[command]git version
+  ##[command]git config --get remote.origin.url
+  ##[command]git clean -ffdx
+> ##[command]git reset --hard HEAD
+  ##[command]git config gc.auto 0
+  ##[command]git config --get-all http.https://repohere
+  ##[command]git config --get-all http.proxy
+> ##[command]git -c http.extraheader="AUTHORIZATION: bearer ***" fetch --tags --prune --progress --no-recurse-submodules origin
+> ##[command]git checkout --progress --force e48a3009f2a0163d102423eef6ffaf7f4c2a2176
+  ##[command]git config http.https://repohere "AUTHORIZATION: bearer ***"
+```
+
+### Open questions
+* Can our tool runner automatically mark the end of output for a command in a reliable way? This way all of the tools that run will get collapsible regions for free.
```

Comment:
It seems like we should be able to emit something after each command. What's the expected syntax for that? Do we inject a `startgroup:foo`, then `command`, then `endgroup:foo`?

Reply:
we could take advantage of implict ending. Just do `##[startgroup]` before command and be done with it. That does mean if there a new command, it will collapse all before, if it's async it can get weird, but am not sure if am too concerned about it.

Reply:
True. @stephenmichaelf is this something we could do automatically in the task-lib?